Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-sorting/ContestSorting.tsx

```diff
@@ -0,0 +1,81 @@
+﻿import React, { useCallback } from 'react';
+import { isNil } from 'lodash';
+import { useContests } from '../../../hooks/use-contests';
+import { ISort } from '../../../common/contest-types';
+import Button, { ButtonSize, ButtonType } from '../../guidelines/buttons/Button';
+import List, { Orientation } from '../../guidelines/lists/List';
+import Heading, { HeadingType } from '../../guidelines/headings/Heading';
+
+import styles from './ContestSorting.module.scss';
+
+interface IContestSortingProps {
+    onSortClick: (sorting: ISort) => void;
+}
+
+const ContestSorting = ({ onSortClick }: IContestSortingProps) => {
+    const { state: { possibleSorting } } = useContests();
+    
+    const handleSortingClick = useCallback(
+        (sortId: number) => {
+            const sorting = possibleSorting.find(({ id }) => sortId === id);
+
+            if (isNil(sorting)) {
+                return;
+            }
+
+            onSortClick(sorting);
+        },
+        [ possibleSorting, onSortClick ],
+    );
+    
+    const renderSortingItem = useCallback(
+        (buttonType: ButtonType, btnClassName: string,name: string, id: number) => (
+            <Button
+                type={buttonType}
+                onClick={() => handleSortingClick(id)}
+                className={btnClassName}
+                text={name}
+                size={ButtonSize.small}
+            />
+        ),
+        [ handleSortingClick ],
+    );
```

Comment:
I see no reason for this to be another function, as it only returns a component.

Reply:
I removed the function.